File: Android Java/Testing/app/src/main/java/com/example/testing/http/Internet.java
package com.example.testing.http;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

import androidx.annotation.NonNull;

import java.net.NetworkInterface;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Collections;
import java.util.List;

public class Internet {
    boolean connected = false;

    public Internet()
    {
        connected = false;

    }
    public boolean checkInternet(Context cgh, Activity activity)
    {

        ConnectivityManager connectivityManager = (ConnectivityManager) cgh.getSystemService(Context.CONNECTIVITY_SERVICE);
        if(connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE).getState() == NetworkInfo.State.CONNECTED ||
                connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI).getState() == NetworkInfo.State.CONNECTED) {
            connected = true;
        }
        else {
            connected = false;
        }
        return connected;
    }


}

File: Android Kotlin/Testing - copia/app/src/main/java/com/example/testing/http/Internet.kt
package com.example.testing.http

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

class Internet {
    var connected = false
    fun checkInternet(cgh: Context, activity: Activity?): Boolean {
        val connectivityManager = cgh.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        connected = if (connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)!!.state == NetworkInfo.State.CONNECTED ||
                connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)!!.state == NetworkInfo.State.CONNECTED) {
            true
        } else {
            false
        }
        return connected
    }

    init {
        connected = false
    }
}